plugin/storage/cassandra/factory.go

```diff
@@ -90,6 +91,9 @@ func NewFactoryWithConfig(
 	}
 	f := NewFactory()
 	f.primaryConfig = &cfg
+	f.Options.Index.Tags = true
+	f.Options.Index.Logs = true
+	f.Options.Index.ProcessTags = true
```

Comment:
this doesn't make sense to me - the objective of this function is to create a factory with externally provided configuration. Instead, we're setting some configuration properties directly - why can they not be set in the caller?

Reply:
made index configurable and added above fields in config file